### PR TITLE
Add join-reordering rules to release note

### PR DESCRIPTION
**Pull request by mkleen**
## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -202,6 +202,13 @@ Data Types
 Administration and Operations
 -----------------------------
 
+- Added optimizer rules for reordering of joins for hash and nested-loop joins.
+  This allows now to control the join-reordering and disable it if desired with
+  the following session settings:
+
+  SET optimizer_reorder_hash_join = false
+  SET optimizer_reorder_nested_loop_join = false
```
Comment (seut) [inline docs/appendices/release-notes/unreleased.rst:210]: I think we should mention that these are experimental settings and such may change in future.

Comment (matriv) [reply docs/appendices/release-notes/unreleased.rst:210]: +1

File: docs/appendices/release-notes/unreleased.rst [hunk 2]
```diff
@@ -202,6 +202,13 @@ Data Types
 Administration and Operations
 -----------------------------
 
+- Added optimizer rules for reordering of joins for hash and nested-loop joins.
+  This allows now to control the join-reordering and disable it, if desired, with
+  session settings. These settings are experimental, and may change in the future:
```
Comment (matriv) [inline docs/appendices/release-notes/unreleased.rst:207]: How about:
```suggestion
  session settings::
  
    SET optimizer_reorder_hash_join = false
    SET optimizer_reorder_nested_loop_join = false

  Note that these settings are experimental, and may change in the future.
```

**Review by matriv — APPROVED** [review]
Thx! Left one more suggestion

File: docs/appendices/release-notes/unreleased.rst [hunk 3]
```diff
@@ -202,6 +202,18 @@ Data Types
...
+  This allows now to control the join-reordering and disable it, if desired, with
+  session settings::
+  
+    SET optimizer_reorder_hash_join = false
+    SET optimizer_reorder_nested_loop_join = false
+
+  Note that these settings are experimental, and may change in the future.
+
+  SET optimizer_reorder_hash_join = false
+  SET optimizer_reorder_nested_loop_join = false
```
Comment (seut) [inline docs/appendices/release-notes/unreleased.rst:215]: Duplicated by mistake?

**Review by seut — APPROVED** [review]
Otherwise, lgtm, thx.

File: docs/appendices/release-notes/unreleased.rst (hunk 3 above)
Comment (mkleen) [reply docs/appendices/release-notes/unreleased.rst]: ups

**Comment by mkleen** [comment]
@Mergifyio refresh

**Comment by mkleen** [comment]
@Mergifyio queue

**Pull request merged**